core/src/wsv.rs

```diff
@@ -1070,37 +1096,24 @@ impl WorldStateView {
         self.events_buffer.borrow_mut().push(event.into());
     }
 
-    /// The same as [`Self::modify_validator_multiple_events`] except closure `f` returns a single [`PermissionValidatorEvent`].
+    /// Get constant view to a *Runtime Permission Validator*
     ///
-    /// # Errors
-    /// Forward errors from [`Self::modify_validators_multiple_events`]
-    pub fn modify_validators<F>(&self, f: F) -> Result<(), Error>
-    where
-        F: FnOnce(&crate::validator::Chain) -> Result<PermissionValidatorEvent, Error>,
-    {
-        self.modify_validators_multiple_events(move |chain| f(chain).map(std::iter::once))
-    }
-
-    /// Get [`crate::validator::Chain`] and pass it to `closure` to modify it
+    /// # Panic
     ///
-    /// # Errors
-    /// Forward errors from `f`
-    pub fn modify_validators_multiple_events<I, F>(&self, f: F) -> Result<(), Error>
-    where
-        I: IntoIterator<Item = PermissionValidatorEvent>,
-        F: FnOnce(&crate::validator::Chain) -> Result<I, Error>,
-    {
-        self.modify_world_multiple_events(|world| {
-            f(&world.validators)
-                .map(|events| events.into_iter().map(WorldEvent::PermissionValidator))
-        })
-    }
+    /// Panics if validator is not initialized.
+    /// Possible only before applying genesis.
+    #[cfg_attr(test, allow(clippy::unused_self))]
+    pub fn validator_view(&self) -> crate::validator::View {
```

Comment:
We can't test real validator in `core` unit tests, because it's passed from outside world. If you'd like to test your own validator you should run integration tests

Reply:
Yes, it's used only to run `iroha_core` unit tests. I'll extend documentation for it